brayns/network/entrypoints/InspectEntrypoint.cpp

```diff
@@ -21,10 +21,103 @@
 
 #include "InspectEntrypoint.h"
 
+namespace
+{
+class SceneInspector
+{
+public:
+    SceneInspector(brayns::Engine &engine, const brayns::Vector2f &inspectPosition)
+        : _engine(engine)
+        , _inspectPosition(inspectPosition)
+    {
+    }
+
+    brayns::InspectResult inspect()
+    {
+        brayns::InspectResult result;
+
+        auto pickResult = _pickOSPRayScene();
+        if (!pickResult.hasHit)
+        {
+            result.hit = false;
+            return result;
+        }
+
+        auto &instance = _findHittedModel(pickResult.instance);
+        auto &model = instance.getModel();
+
+        result.hit = true;
+        result.model_id = instance.getID();
+
+        auto &ospHitPosition = pickResult.worldPosition;
+        result.position = brayns::Vector3f(ospHitPosition[0], ospHitPosition[1], ospHitPosition[2]);
+
+        const auto &hitPosition = result.position;
+        auto geometricModel = pickResult.model;
+        auto primitiveID = pickResult.primID;
+
+        brayns::JsonObject data;
+        model.onInspect(hitPosition, geometricModel, primitiveID, data);
+
+        auto &metadata = result.metadata;
+        metadata = brayns::JsonValue(data);
```

Comment:
The cast is really necessary ? Normally Poco::Var has a template constructor that accepts anything.